Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,135 @@
+    func fetchComic() {
+        fetchingService.fetchComic(comicType: .latestComic) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                print(error.localizedDescription)
+            }
+        }
+    }
+    
+    func fetchPreviousComic() {
+        print("VM fetchPreviousComic: number of current comic: \(comic.num)")
+        fetchingService.fetchComic(comicType: .specificComic(comic.num - 1)) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingPreviousComicAlert = true
+                }
+                print("ComicsVM, fetching previous comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func fetchNextComic() {
+        fetchingService.fetchComic(comicType: .specificComic(comic.num + 1)) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingNextComicAlert = true
+                }
+                print("ComicsVM, fetching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func toggleDescription() {
+        isShowingExplanation.toggle()
+    }
+    
+    func toggleInfo() {
+        isShowingInfo.toggle()
+    }
+    
+    func searchComic(searchNum: String) {
+        guard isShowingSearch else {
+            isShowingSearch = true
+            return
+        }
+        // TODO: error warning if the input is not int
+        fetchingService.fetchComic(comicType: .specificComic(Int(searchNum) ?? 1)) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    self.isSaved = self.isSaved(comic: comic)
+                    self.isShowingSearch = false
+                }
+            case .failure(let error):
+                print("ComicsVM, searching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func saveAsFavourite(comic: Comic) {
+        guard !isSaved else {
+            print("already saved")
+            return
+        }
+        DispatchQueue.main.async {
+            self.isSaved = self.isSaved(comic: comic)
+        }
+        userDataService.addComicToFavorites(comic: comic)
+    }
+    
+    func getFavoriteComics() {
+        guard let comics = userDataService.favoriteComics() else {
+            return
+        }
+        for comic in comics {
+            print("ComicVM, getting favorite comics: \(comic.title)")
+        }
+    }
+    
+    private func isSaved(comic: Comic) -> Bool {
```

Comment:
Something I forgot to mention in the previous review: A method like `isSaved` should really be exposed by `UserDataService` if checking requires several lines of code. Anyone should be able to query UserDataService to quickly see whether something is saved, without having to re-implement all of this logic for fetching the entire array and looking through it.